### DISCUSSION TITLE
monorepo with different abl projects

### DISCUSSION THREAD

**Discussion by jmls** [vscode-abl/vscode-abl#68] [General]
Is it possible to configure the plugin to handle multiple ABL projects within a monorepo ?

Each project would have it's own propath, db connections etc 

I'm not sure that profiles fit the bill, as it seems to be orientated for different abl versions. Or can they be "manipulated" to support a monorepo ?

**Comment by kenherring-rkt** [comment]
This would be very helpful for us as well.  Ideally having configuration similar to the sonar "submodules" work would be fantastic.

**Comment by gquerret** [comment]
Assuming your monorepo structure is like that:
```
root
 \_ oe_project1
     \_ openedge-project.json
 \_ oe_project2
     \_ openedge-project.json
 \_ node_project1
 \_ ...
```

You can have a `.code-workspace` file in the root directory, which contains references to `oe_project1` and `oe_project2`. The language server will then try to open the configuration file of each project of the workspace.

**Reply by PeterJudgeZA** [reply]
FWIW this is an approach we take.

**Comment by gquerret** [comment]
@jmls @kenherring-rkt Did you try opening vscode with the workspace configuration ?

**Reply by kenherring-rkt** [reply]
Didn't love how it worked for us - mostly because we have files for different "projects" comingled in directories.  Probably not worth fixing and just something we have to deal with.
